Devnet scripting (#124)
carbonable-labs/carbon-protocol-v3: Contributing.md

#### Compile

To compile the project, run:

```bash
scarb build
```

#### Set you env

> Follow the .env.template

Reviewer: Could you push the template?

```bash
source .env
```

#### Code Style

To format the code, run:

```bash
scarb fmt
```

#### Testing

To run tests (using Starknet-Foundry), use:

```bash
scarb test
```

To run a specific test:

```bash
scarb test <name_of_the_test>
```

#### Devnet

To generata an accound see :
[this page](https://docs.starknet.io/quick-start/set-up-an-account/)

```bash
cargo install starknet-devnet
```

To run a new node

```bash
starknet-devnet
```

Fork existing chain with flag

```bash
--fork-network $FORK_URL
```

for further options

```bash
starknet-devnet --help
```

#### Deploy

To deploy (on testnet/mainnet/devnet), use:

> 1 Check available contract

Reviewer: ```suggestion
1. Check available contract
```

```bash
bash scripts/declare.sh
```

2 Declare contract -> it gives you a class_hash

```bash

> bash scripts/declare.sh <name>

Reviewer: ```suggestion
bash scripts/declare.sh <contract_name>
```

```

> 3 Deploy contract (order mock (if needed), project, minter, offsetter, minter)

Reviewer: ```suggestion
3. Deploy contracts
If you want to simulate the whole infra, please deploy in this order: mock_stable, project, minter, offsetter
```

```bash
bash scripts/deploy_<name>.sh <ClASS_HASH> <...ARGS>
```

> You can batch 2&3
> 
> ```bash
> bash scripts/declare.sh <name> | tail -n 1 |read HASH && bash scripts/deploy_<name>.sh <HASH> <...ARGS>
> ```

Reviewer: ```suggestion
```

delete this part temporarly, as it's not working for every config systems
